### PR TITLE
Specify OAuth Redirection URI in ENV

### PR DESCRIPTION
**Pull request by nomadicoder**
Currently we are hardcoding the Redirect URI in code that's in source control. This should take care of that problem code by specifying the URI as an environment variable. If not specified, URI defaults to `http://localhost:3000/accounts/auth/google_oauth2`.

### CODE REVIEW

**Comment by nomadicoder** [comment]
@chad - I have updated the .bashrc on the production server. We need to test this in production.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 module ApplicationHelper
+  def google_oauth_redirect_uri
+    ENV.fetch('GOOGLE_OAUTH_REDIRECT_URI') { 'http://localhost:3000/accounts/auth/google_oauth2' }
```
Comment (bibliotechy) [inline app/helpers/application_helper.rb]: Why not just use `"#{root_path}accounts/auth/google_oauth2"`?

Comment (nomadicoder) [reply app/helpers/application_helper.rb]: Can you redirect to a file path if you want a URI which is required by Google OAuth?

Comment (nomadicoder) [reply app/helpers/application_helper.rb]: We can use `Socket.gethostbyname(Socket.gethostname).first` in production. See: https://stackoverflow.com/questions/151545/how-can-i-get-the-fqdn-of-the-current-host-in-ruby

We can use `localhost:3000` when we're in development.

**Comment by nomadicoder** [comment]
@bibliotechy Get FQDN without using an environment variable.

**Comment by bibliotechy** [comment]
What I really should have pointed you to was using the builtin Rails route helpers [Path and URL Helpers](https://guides.rubyonrails.org/routing.html#path-and-url-helpers)
> Each of these helpers has a corresponding _url helper (such as photos_url) which returns the same path prefixed with the current host, port and path prefix.

So, in this case, we can use `account_google_oauth2_omniauth_authorize_url` directly in the template, with no need for an additional helper.

**Comment by nomadicoder** [comment]
@bibliotechy This is how it's supposed to be done.

File: README.md [hunk 2]
```diff
@@ -13,9 +13,15 @@ Fortytude is the Temple University Library's website built on Ruby on Rails.
 ```
 export GOOGLE_OAUTH_CLIENT_ID="Google client ID goes here"
 export GOOGLE_OAUTH_SECRET="Google OAuth secret goes here"
+export GOOGLE_OAUTH_REDIRECT_URI="Google OAuth Redirection URI goes here -- Set below"
```
Comment (bibliotechy) [inline README.md]: This is no longer need, correct?

Comment (bibliotechy) [reply README.md]: @nomadicoder ping?

**Comment by nomadicoder** [comment]
@bibliotechy We'll get this right some day.  Now using URL helper for the redirect.

**Comment by nomadicoder** [comment]
@bibliotechy README fixed and just has the punctuation fix.

**Pull request merged**